base/logging_bucket.go

```diff
@@ -1,230 +1,217 @@
 package base
 
 import (
+	"context"
 	"expvar"
+	"sync"
 	"time"
 
 	sgbucket "github.com/couchbase/sg-bucket"
 )
 
 // A wrapper around a Bucket that transparently adds logging of all the API calls.
 type LoggingBucket struct {
-	bucket Bucket
+	bucket     Bucket
+	logCtx     context.Context
+	logCtxOnce sync.Once
+}
+
+func (b *LoggingBucket) ctx() context.Context {
+	b.logCtxOnce.Do(func() {
+		b.logCtx = bucketCtx(context.Background(), b)
+	})
+	return b.logCtx
+}
+
+func (b *LoggingBucket) log(start time.Time, args ...interface{}) {
```

Comment:
This is a nice approach to minimize maintenance on the logging calls.  We don't care about performance overhead when doing bucket-level logging so we shouldn't need to worry about GetCallersName, I assume?

Reply:
GetCallersName without file/line numbers isn't too expensive (~685ns/op - I've just added a benchmark for this), but you're right - using LoggingBucket isn't performance critical, and is normally disabled.